api/src/main/java/org/eclipse/microprofile/reactive/messaging/connector/OutgoingConnectorFactory.java

```diff
@@ -0,0 +1,109 @@
+ * implementation. It must match the class returned by the {@link #type()} method. This is how a reactive messaging
+ * implementation looks for the specific {@link OutgoingConnectorFactory} required for a channel. In the previous
+ * configuration, the reactive messaging implementation would need to find the {@link OutgoingConnectorFactory} returning
+ * the {@code i.e.m.reactive.messaging.impl.kafka.Kafka} class as the result of its {@link #type()} method to create the
+ * {@code my-channel} <em>subscriber</em>. Note that if the connector cannot be found, the deployment must be failed
+ * with a {@link javax.enterprise.inject.spi.DeploymentException}.
+ * <p>
+ * The {@link #getSubscriberBuilder(Config)} is called for every <em>channel</em> that needs to be created. The
+ * {@link Config} object passed to the method contains a subset of the global configuration, and with the prefixes removed.
+ * So for the previous configuration, it would be:
+ * <pre>
+ * bootstrap-servers = localhost:9092
+ * topic = my-topic
+ * </pre>
+ * <p>
+ * So the connector implementation can retrieve the value with {@link Config#getValue(String, Class)} and
+ * {@link Config#getOptionalValue(String, Class)}.
+ * <p>
+ * If the configuration is invalid, the {@link #getSubscriberBuilder(Config)} method must throw an
+ * {@link IllegalArgumentException}, caught by the reactive messaging implementation and triggering a failure in the
+ * deployment.
+ * <p>
+ * Note that a Reactive Messaging implementation must support the configuration format described here. Implementations
+ * are free to provide additional support for other approaches.
+ */
+public interface OutgoingConnectorFactory {
+
+    /**
+     * Gets the {@link MessagingProvider} class associated with this {@link OutgoingConnectorFactory}.
+     * Note that the {@link MessagingProvider} is a user-facing interface used in the configuration.
+     *
+     * @return the {@link MessagingProvider} associated with this {@link OutgoingConnectorFactory}. Must not be
+     * {@code null}. Returning {@code null} triggers a deployment failure.
+     */
+    Class<? extends MessagingProvider> type();
+
+    /**
+     * Creates a <em>channel</em> for the given configuration. The given configuration {@code type} attribute matches the
+     * {@link #type()}.
+     * <p>
+     * Note that the connection to the <em>transport</em> or <em>broker</em> is generally postponed until the
+     * subscription.
+     *
+     * @param config the configuration, never {@code null}
+     * @return the created {@link SubscriberBuilder}, must not be {@code null}.
+     * @throws IllegalArgumentException if the configuration is invalid.
+     * @throws NoSuchElementException if the configuration does not contain an expected attribute.
+     */
+    SubscriberBuilder<? extends Message, Void> getSubscriberBuilder(Config config);
```

Comment:
I don't think we should ask a config object to be passed over.

Reply:
This has been discussed during the hangout.